### PR TITLE
Add system properties to configure Jackson's stream read constraints

### PR DESCRIPTION
**Pull request by edmocosta**
## Release notes

Added `jvm.options` properties to configure the Jackson read constraints defaults (Maximum Number value length, Maximum String value length, and Maximum Nesting depth).

## What does this PR do?

This PR adds 3 new custom system properties into the `jvm.options`. Those new properties are used to fine-tune the default Jackson stream read constraints values, which are used by Jackson to guard against malicious input by preventing processing of too big inputs. The added properties can be found [here](https://github.com/elastic/logstash/pull/15720/files#diff-d592447cb7aefcc8d1929fd64fa5dcb40a77e476e221f9690a88e01a002f264bR74-R88).

The intention of early set the Jackson defaults during the startup (`runner.rb`) was to ensure that all Logstash's functionalities that rely on Jackson are using those settings values.

## Why is it important/What is the impact to the user?

Although the Jackson's defaults works for the majority of users, it might be too restrictive - and a blocker - for pipelines receiving big payload requests or with a deeper level of nested objects. 

Having the possibility of tuning those properties is essential to unblock users with that use-case and for a more accurate guard against malicious inputs.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files (and/or docker env variables)
- [x] I have added tests that prove my fix is effective or that my feature works

## How to test this PR locally

- Set any property value to a lower/higher value, e.g.: `-Dlogstash.jackson.stream-read-constraints.max-string-length=2`
- Configure and run a pipeline with a `beats` input
- Send any data to this pipeline that exceeds the configured property value
- Check the logs for: `com.fasterxml.jackson.core.exc.StreamConstraintsException: String length (24) exceeds the maximum length (2)`

## Related issues

- Closes https://github.com/elastic/ingest-dev/issues/2638

### CODE REVIEW

File: config/jvm.options [hunk 1]
```diff
@@ -69,4 +69,20 @@
...
+-Dlog4j2.isThreadContextMapInheritable=true
+
+# FasterXML/jackson defaults
+#
+# Sets the maximum string length (in chars or bytes, depending on input context).
+# This limit is not exact and an exception will happen at sizes greater than this limit.
+# Some text values that are a little bigger than the limit may be treated as valid but no
+# text values with sizes less than or equal to this limit will be treated as invalid.
+# This value should be higher than `logstash.jackson.stream-read-constraints.max-number-length`.
+#-Dlogstash.jackson.stream-read-constraints.max-string-length=20000000
```
Comment (jsvd) [inline config/jvm.options:81]:
```suggestion
# the jackson library defaults to 20000000 or 20MB, whereas Logstash defaults to 200MB or 200000000 characters
-Dlogstash.jackson.stream-read-constraints.max-string-length=200000000
```

File: config/jvm.options [hunk 2]
```diff
@@ -69,4 +69,20 @@
...
+#
+# Sets the maximum number length (in chars or bytes, depending on input context).
+#-Dlogstash.jackson.stream-read-constraints.max-number-length=1000
```
Comment (jsvd) [inline config/jvm.options:84]:
```suggestion
# jackson library defaults to 1000, whereas Logstash defaults to 10000
-Dlogstash.jackson.stream-read-constraints.max-number-length=10000
```

**Review by jsvd — COMMENTED** [review]
just tweaking the defaults here: raising the number and message size by 10x. especially message size 20mb will be too low

**Review by jsvd — APPROVED** [review]
LGTM

**Comment by edmocosta** [comment]
@logstashmachine backport 8.12

**Pull request merged**